dna/clutter/clutter.js

```diff
@@ -1,174 +1,317 @@
+var FIRST_NAME = "firstName";
 
-function getProperty(name) {            // The definition of the function you intend to expose
-    return property(name);              // Retrieves a property of the holochain from the DNA (e.g., Name, Language
+function getProperty(name) {
+  // The definition of the function you intend to expose
+  return property(name); // Retrieves a property of the holochain from the DNA (e.g., Name, Language
 }
-function appProperty(name) {            // The definition of the function you intend to expose
-    if (name == "Agent_Handle") {
-      debug("Agent_Handle");
-      debug(getHandle(App.Key.Hash));
-      return getHandle(App.Key.Hash);
+
+/**
+ * @param data is a string representing a firstName
+ * @return nameHash is the hash of the firstName
+ **/
+function setFirstName(data) {
+  var nameHash;
+  try {
+    // Check if name has been set and update if so.
+    var links = getLinks(App.Agent.Hash, FIRST_NAME, { Load: true });
+
+    if (links.length > 0) {
+      nameHash = update(FIRST_NAME, data, links[0].Hash);
+    } else {
+      // Otherwise add it for the first time:
+      nameHash = commit(FIRST_NAME, data);
+      // On the DHT, put a link from my hash to the hash of firstName
+      var me = App.Agent.Hash;
+      commit("profile_links", {
+        Links: [{ Base: me, Link: nameHash, Tag: FIRST_NAME }]
+      });
     }
-    if (name == "App_Agent_String")  {return App.Agent.String;}
-    if (name == "App_Key_Hash")   {return   App.Key.Hash;}
-    if (name == "App_DNA_Hash")   {return   App.DNA.Hash;}
-    return "Error: No App Property with name: " + name;
+  } catch (exception) {
+    debug(exception);
+  }
+  return data;
```

Comment:
In the function comments, it says it will return the hash, but it returns the raw input instead. Best to reconcile them, one way or another.

Reply:
done